fenix/app/src/main/java/org/mozilla/fenix/components/StoreProvider.kt

```diff
@@ -10,19 +10,30 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.ViewModelStoreOwner
 import androidx.lifecycle.get
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CoroutineScope
 import mozilla.components.lib.state.Store
 
 /**
  * Generic ViewModel to wrap a State object for state restoration.
  *
- * @property store [Store] instance attached to [ViewModel].
+ * @param createStore Function that creates a [Store] instance that receives [CoroutineScope] param
+ * that's tied to the lifecycle of the [StoreProvider] i.e it will cancel when
+ * [StoreProvider.onCleared] is called.
  */
 class StoreProvider<T : Store<*, *>>(
-    val store: T,
+    createStore: (CoroutineScope) -> T,
 ) : ViewModel() {
 
+    @VisibleForTesting
+    val store: T = createStore(viewModelScope)
```

Comment:
Since you're adding `@VisibileForTesting` here, presumably because this is used only for tests so far aside from the return in `get()`, we should add the `internal` visibility qualifier too then.

```suggestion
    @VisibleForTesting
    internal val store: T = createStore(viewModelScope)
```